server/jetstream_cluster.go

```diff
@@ -2361,6 +2361,8 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 			return
 		}
 
+		// Make sure all pending data is flushed before allowing snapshots.
+		mset.flushAllPending()
```

Comment:
If the stream is us9ing small blocks like KV, the number of blocks could be very large. We will cycle through each one for every snapshot, acquiring the lock etc..

Reply:
Discussed this privately as well, but have changed it so that we don't loop over all `fs.blks`, instead we only need to check and flush `fs.lmb`.